### PR TITLE
pfblockerng: parse bracketed IPv6 literals and strip UTF-8 BOM in DNSBL feeds

### LINKED ISSUE #938: DNSBL parser: bracketed IPv6 literals rejected; UTF-8 BOM defeats comment stripping
Two DNSBL parser gaps produce recurring `dnsbl_parsed_error.log` noise on every update of common feeds (observed in production against StevenBlack and Dandelion Sprout's Anti-Malware hosts lists):

## 1. Bracketed IPv6 literals are never parsed

Dandelion Sprout's hosts-file list contains AdGuard-style rules whose host part is a bracketed IPv6 literal, e.g.:

```text
://[2604:2dc0:100:4ed8::]
://[2604:2dc0:104:da00::]
://[2600:3c06::f03c:95ff:fed9:ce5e]
```

After the scheme strip (`pfb_dnsbl_scheme_line`), the line remains `[2604:2dc0:100:4ed8::]` — `is_ipaddrv6()` rejects the brackets, the domain filter then rejects it too, and the line lands in the parse-error log on every feed update.

The IPv6 collection plumbing already exists end-to-end (`$domain_data_ip6`, `validate_ipv6()`, the DNSBL-IP extraction feature; CSV feeds already collect IPv6 via `is_ipaddrv6($csvline[0])`). The generic line parser is only missing the bracket strip (and tolerance for the bare `://`-prefixed rule form) before the existing `is_ipaddrv6($line)` check at `pfblockerng.inc:16737`.

Log evidence (raw `dnsbl_parsed_error.log`):

```text
06/23/26 08:55:45,DandelionSprouts,://[2604:2dc0:100:4ed8::],://[2604:2dc0:100:4ed8::]
07/4/26 00:30:21,DandelionSprouts,://[2600:3c06::f03c:95ff:fed9:ce5e],://[2600:3c06::f03c:95ff:fed9:ce5e]
```

## 2. UTF-8 BOM defeats comment stripping

The same feed starts with a BOM followed by a comment header. The BOM survives into the first line, so the `#` comment strip does not recognise it and the header is error-logged on every update:

```text
07/7/26 18:08:24,DandelionSprouts,Title:,ï»¿# Title: ð Dandelion Sprout's Anti-Malware List (Â«hostsÂ» file version)
```

A UTF-8 BOM (`EF BB BF`) at the start of the feed (first line) should be stripped before line classification.

## Expected outcome

- Bracketed IPv6 rule lines feed the existing DNSBL-IP v6 extraction (when enabled) instead of the parse-error log, consistent with how bare IPv6 literals and CSV-feed IPv6 entries are already collected.
- BOM-prefixed comment lines are recognised as comments and skipped silently.
- Neither pattern appears in `dnsbl_parsed_error.log` afterwards.

Out of scope (tracked separately if desired): `!#if`/`!#endif` AdGuard directives inside hosts-format feeds, `ip6-loopback` missing from the `$dnsbl_skip` boilerplate list.

### PR DESCRIPTION
**Pull request by andrebrait**
Fixes #938.

Two generic-parser (non-ABP) DNSBL gaps that spam `dnsbl_parsed_error.log` on every update of common feeds (StevenBlack, Dandelion Sprout's Anti-Malware list):

1. **Bracketed IPv6 literals** (`://[2604:2dc0:100:4ed8::]`) survive scheme/port stripping unrecognised — `is_ipaddrv6()` rejects the brackets, the domain filter rejects the remainder, and the line is error-logged instead of feeding the existing DNSBL-IP v6 extraction. New `pfb_dnsbl_unbracket_ip6()` unwraps a `[`…`]` pair **only when the inner text is a valid IPv6 literal**; everything else (`[not:an:ip]`, `[]`, bracketed IPv4) flows to domain validation and the parse-error log exactly as before.
2. **A UTF-8 BOM** on a feed's first line defeats the `#` comment check, so a BOM-led `# Title: …` header is error-logged on every update. New `pfb_dnsbl_strip_bom()` strips exactly one leading BOM before first-char classification (same class of handling as `pfb_text_sanity()` and `pfb_dnsbl_is_abp_header()`).

Both helpers sit beside `pfb_dnsbl_strip_scheme()` and follow its style. Deliberately untouched: the ABP branch (own IP handling), hosts-format extraction, ADR-22 strict-mode scheme semantics (strict still rejects the empty-scheme `://[v6]` form; lenient — the migrated default — extracts), CSV cases, `$dnsbl_skip`, and `!`-directive handling (out of scope per the issue).

Tests: `PfbDnsblStripBomTest` (6 rows) + `PfbDnsblUnbracketIp6Test` (13 rows incl. the port-strip composition pinning call-site order and the zone-id literal's probed behaviour) — 19 tests red on pre-change code (undefined function), green after; full suite 2376 passing; phpstan + phpcs clean.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

https://claude.ai/code/session_01PfJUsCQkpFZxXufncGUS2f

## Summary by CodeRabbit

* **New Features**
  * Improved DNS feed line handling by automatically cleaning up leading BOM characters and recognizing bracketed IPv6 literals more reliably.
* **Bug Fixes**
  * Prevented BOM-prefixed comments and data from being misread during parsing.
  * Avoided valid bracketed IPv6 entries being treated as invalid domain data or logged as parse errors.
* **Tests**
  * Added coverage for BOM stripping and IPv6 unbracketing edge cases, including malformed inputs and special IPv6 formats.

### CODE REVIEW

**Comment by andrebrait** [comment]
@coderabbitai review

---
🤖 Generated by [Claude Code](https://claude.com/claude-code), posted via @andrebrait's account on their behalf.

File: src/usr/local/pkg/pfblockerng/pfblockerng.inc [hunk 1]
```diff
@@ -6469,6 +6469,38 @@ function pfb_dnsbl_strip_scheme(string $line, bool $strict = FALSE): string|fals
 }
 
 
+// Issue #938: strip a leading UTF-8 BOM (EF BB BF) -- encoding metadata, not content,
+// the same class of noise pfb_text_sanity() strips up front (line ~1054) and
```
[the remark this answers is not in the document]
Comment (andrebrait) [reply src/usr/local/pkg/pfblockerng/pfblockerng.inc]: Applied in acce0e64 — dropped the hardcoded "(line ~1054)" reference; the comment now cites pfb_text_sanity() by name only.
---
🤖 Generated by [Claude Code](https://claude.com/claude-code), posted via @andrebrait's account on their behalf.

File: tests/php/PfbDnsblUnbracketIp6Test.php [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+			pfb_dnsbl_unbracket_ip6(pfb_strip_trailing_port('[2604:2dc0::]:443'))
+		);
+	}
+
+	public function testZoneIdLiteralFollowsIsIpaddrv6Verbatim(): void
+	{
+		// Observed truth (run, not guessed): the pfSense is_ipaddrv6() double treats
+		// 'fe80::1%em0' as a valid link-local literal (zone-id retained in the address
+		// grammar for link-local addresses), so the bracket wrapper unwraps it verbatim
+		// -- the helper never itself strips the zone id, it only defers to is_ipaddrv6().
```
[the remark this answers is not in the document]
Comment (andrebrait) [reply tests/php/PfbDnsblUnbracketIp6Test.php]: Applied in acce0e64 — the rationale now states the double's actual mechanism (the %zone suffix is stripped during validation, per pfsense_doubles.php) while keeping the observable contract: the helper returns the original bracketed literal verbatim and defers validity entirely to is_ipaddrv6().
---
🤖 Generated by [Claude Code](https://claude.com/claude-code), posted via @andrebrait's account on their behalf.

**Comment by andrebrait** [comment]
## Review audit trail (pre-merge)

**Review shape:** single Claude Sonnet 5 adversarial reviewer at effort `xhigh` (PR size: 1 source file + 2 test files, ~90 added lines — below the fanout threshold).

**Reviews received:**

- **Claude Sonnet 5 adversarial review** — no blocking findings. Independently re-ran the full PHPUnit suite (2376/2376), PHPStan (0 errors), PHPCS incl. both custom sniffs (clean), and re-executed the red→green proof itself (19 real `undefined function` errors on pre-change source; 19/19 green after; clean restore). Verified every out-of-scope item untouched (ABP branch, hosts extraction, ADR-22 strict-mode empty-scheme rejection, CSV switch, `$dnsbl_skip`, `!`-lines, `pfb_parsed_fail`) and probed hostile inputs beyond the acceptance rows (empty/whitespace/oversized/multi-bracket/UTF-16-BOM inputs — no crashes, all correctly unchanged).
- **GitHub Copilot** — 2 findings, both applied.
- **CodeRabbit** — did NOT review: rate-limited twice (initial notice, then again after the one permitted `@coderabbitai review` nudge; stated resume 54 minutes > the 5-minute rule). Skipped per flow policy — this PR was **not** reviewed by CodeRabbit.
- **Snyk** — status `error` ("Code test limit reached"): the scan did not run. Noted as a skipped scan, not a clean pass.

**Findings ledger:**

1. Copilot — drifting hardcoded line reference in `pfb_dnsbl_strip_bom()` comment → **fixed@acce0e64**
2. Copilot — zone-id test rationale misstated the `is_ipaddrv6()` double's mechanism (it strips `%zone` during validation) → **fixed@acce0e64**
3. Sonnet (nitpick) — UTF-16 BOMs (`FE FF`/`FF FE`) still unhandled → **skipped**: issue #938 is scoped to the UTF-8 BOM; the whole parser is byte-oriented and a UTF-16 feed is a different-encoding problem end to end (reviewer concurs, no fix suggested)
4. Sonnet (nitpick) — ADR-21 `||`-anchor short-circuit runs before the BOM strip, so a hypothetical BOM-prefixed anchor on a feed's first line would take the generic path → **skipped**: BOM occurs only on line 1 and a real feed's line 1 is a header, never an anchor rule; not an acceptance row (reviewer concurs, no fix suggested)
5. Sonnet (nitpick) — `[::ffff:1.2.3.4]` unwraps and routes to the IPv6 bucket → **skipped**: identical routing to what a bare `::ffff:1.2.3.4` line already gets today via the same `is_ipaddrv6()` collection branch — consistent with pre-existing dual-stack handling, not a regression (empirically verified by the reviewer)

Gates re-run after the review-fix commit (`acce0e64`): `php -l` clean, PHPUnit 2376/2376, PHPStan 0 errors, PHPCS clean.

---
🤖 Generated by [Claude Code](https://claude.com/claude-code), posted via @andrebrait's account on their behalf.

**Comment by andrebrait** [comment]
**Ledger correction** (per the review-finding policy updated on `devel` mid-flow): findings 3 and 4 of the audit above — confirmed-real but out-of-scope edges (UTF-16 BOMs unhandled; ADR-21 anchor check precedes the BOM strip) — are reclassified from *skipped* to **deferred → #946**. Finding 5 remains skipped (no defect: routing identical to pre-existing bare-literal behaviour, empirically verified).

---
🤖 Generated by [Claude Code](https://claude.com/claude-code), posted via @andrebrait's account on their behalf.
